### ISSUE TITLE
Node 12 support

### ISSUE THREAD

**Issue by alexcastillo** [avz/node-posix-clock#2]
Hi,

Thanks for this project!

After upgrading to Node 12, I noticed `node-posix-clock` is incompatible. Here are the errors I'm getting:

```
> node-gyp rebuild

  CXX(target) Release/obj.target/posix-clock/src/posix-clock.o
../src/posix-clock.cpp:32:53: error: too few arguments to function call, single argument
      'context' was not specified
        clockid_t clockId = (clockid_t)info[0]->Int32Value();
                                       ~~~~~~~~~~~~~~~~~~~ ^
/Users/alexcastillo/.node-gyp/12.2.0/include/node/v8.h:2569:3: note: 'Int32Value' declared here
  V8_WARN_UNUSED_RESULT Maybe<int32_t> Int32Value(Local<Context> context) const;
  ^
/Users/alexcastillo/.node-gyp/12.2.0/include/node/v8config.h:347:31: note: expanded from macro
      'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
../src/posix-clock.cpp:47:2: warning: 'Set' is deprecated: Use maybe version
      [-Wdeprecated-declarations]
        AVZ_FILL_TIMESPEC(obj, ts.tv_sec, ts.tv_nsec);
        ^
../src/posix-clock.cpp:15:13: note: expanded from macro 'AVZ_FILL_TIMESPEC'
                (target)->Set(Nan::New("sec").ToLocalChecked(), Nan::New<Number>(sec)); \
                          ^
/Users/alexcastillo/.node-gyp/12.2.0/include/node/v8.h:3358:3: note: 'Set' has been explicitly
      marked deprecated here
  V8_DEPRECATE_SOON("Use maybe version",
  ^
/Users/alexcastillo/.node-gyp/12.2.0/include/node/v8config.h:322:29: note: expanded from macro
      'V8_DEPRECATE_SOON'
  declarator __attribute__((deprecated(message)))
                            ^
../src/posix-clock.cpp:47:2: warning: 'Set' is deprecated: Use maybe version
      [-Wdeprecated-declarations]
        AVZ_FILL_TIMESPEC(obj, ts.tv_sec, ts.tv_nsec);
        ^
../src/posix-clock.cpp:16:13: note: expanded from macro 'AVZ_FILL_TIMESPEC'
                (target)->Set(Nan::New("nsec").ToLocalChecked(), Nan::New<Number>(nsec));
                          ^
/Users/alexcastillo/.node-gyp/12.2.0/include/node/v8.h:3358:3: note: 'Set' has been explicitly
      marked deprecated here
  V8_DEPRECATE_SOON("Use maybe version",
  ^
/Users/alexcastillo/.node-gyp/12.2.0/include/node/v8config.h:322:29: note: expanded from macro
      'V8_DEPRECATE_SOON'
  declarator __attribute__((deprecated(message)))
                            ^
../src/posix-clock.cpp:60:53: error: too few arguments to function call, single argument
      'context' was not specified
        clockid_t clockId = (clockid_t)info[0]->Int32Value();
                                       ~~~~~~~~~~~~~~~~~~~ ^
/Users/alexcastillo/.node-gyp/12.2.0/include/node/v8.h:2569:3: note: 'Int32Value' declared here
  V8_WARN_UNUSED_RESULT Maybe<int32_t> Int32Value(Local<Context> context) const;
  ^
/Users/alexcastillo/.node-gyp/12.2.0/include/node/v8config.h:347:31: note: expanded from macro
      'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
../src/posix-clock.cpp:75:2: warning: 'Set' is deprecated: Use maybe version
      [-Wdeprecated-declarations]
        AVZ_FILL_TIMESPEC(obj, ts.tv_sec, ts.tv_nsec);
        ^
../src/posix-clock.cpp:15:13: note: expanded from macro 'AVZ_FILL_TIMESPEC'
                (target)->Set(Nan::New("sec").ToLocalChecked(), Nan::New<Number>(sec)); \
                          ^
/Users/alexcastillo/.node-gyp/12.2.0/include/node/v8.h:3358:3: note: 'Set' has been explicitly
      marked deprecated here
  V8_DEPRECATE_SOON("Use maybe version",
  ^
/Users/alexcastillo/.node-gyp/12.2.0/include/node/v8config.h:322:29: note: expanded from macro
      'V8_DEPRECATE_SOON'
  declarator __attribute__((deprecated(message)))
                            ^
../src/posix-clock.cpp:75:2: warning: 'Set' is deprecated: Use maybe version
      [-Wdeprecated-declarations]
        AVZ_FILL_TIMESPEC(obj, ts.tv_sec, ts.tv_nsec);
        ^
../src/posix-clock.cpp:16:13: note: expanded from macro 'AVZ_FILL_TIMESPEC'
                (target)->Set(Nan::New("nsec").ToLocalChecked(), Nan::New<Number>(nsec));
                          ^
/Users/alexcastillo/.node-gyp/12.2.0/include/node/v8.h:3358:3: note: 'Set' has been explicitly
      marked deprecated here
  V8_DEPRECATE_SOON("Use maybe version",
  ^
/Users/alexcastillo/.node-gyp/12.2.0/include/node/v8config.h:322:29: note: expanded from macro
      'V8_DEPRECATE_SOON'
  declarator __attribute__((deprecated(message)))
                            ^
../src/posix-clock.cpp:89:53: error: too few arguments to function call, single argument
      'context' was not specified
        clockid_t clockId = (clockid_t)info[0]->Int32Value();
                                       ~~~~~~~~~~~~~~~~~~~ ^
/Users/alexcastillo/.node-gyp/12.2.0/include/node/v8.h:2569:3: note: 'Int32Value' declared here
  V8_WARN_UNUSED_RESULT Maybe<int32_t> Int32Value(Local<Context> context) const;
  ^
/Users/alexcastillo/.node-gyp/12.2.0/include/node/v8config.h:347:31: note: expanded from macro
      'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
../src/posix-clock.cpp:96:34: error: too few arguments to function call, single argument
      'context' was not specified
        int flags = info[1]->Int32Value();
                    ~~~~~~~~~~~~~~~~~~~ ^
/Users/alexcastillo/.node-gyp/12.2.0/include/node/v8.h:2569:3: note: 'Int32Value' declared here
  V8_WARN_UNUSED_RESULT Maybe<int32_t> Int32Value(Local<Context> context) const;
  ^
/Users/alexcastillo/.node-gyp/12.2.0/include/node/v8config.h:347:31: note: expanded from macro
      'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
../src/posix-clock.cpp:107:36: error: no matching member function for call to 'ToObject'
        Local<Object> objSleep = info[2]->ToObject();
                                 ~~~~~~~~~^~~~~~~~
/Users/alexcastillo/.node-gyp/12.2.0/include/node/v8.h:2532:44: note: candidate function not
      viable: requires single argument 'context', but no arguments were provided
  V8_WARN_UNUSED_RESULT MaybeLocal<Object> ToObject(
                                           ^
/Users/alexcastillo/.node-gyp/12.2.0/include/node/v8.h:2546:35: note: candidate function not
      viable: requires single argument 'isolate', but no arguments were provided
                    Local<Object> ToObject(Isolate* isolate) const);
                                  ^
../src/posix-clock.cpp:108:36: warning: 'Get' is deprecated: Use maybe version
      [-Wdeprecated-declarations]
        Local<Value> secValue = objSleep->Get(Nan::New("sec").ToLocalChecked());
                                          ^
/Users/alexcastillo/.node-gyp/12.2.0/include/node/v8.h:3412:3: note: 'Get' has been explicitly
      marked deprecated here
  V8_DEPRECATE_SOON("Use maybe version", Local<Value> Get(Local<Value> key));
  ^
/Users/alexcastillo/.node-gyp/12.2.0/include/node/v8config.h:322:29: note: expanded from macro
      'V8_DEPRECATE_SOON'
  declarator __attribute__((deprecated(message)))
                            ^
../src/posix-clock.cpp:109:37: warning: 'Get' is deprecated: Use maybe version
      [-Wdeprecated-declarations]
        Local<Value> nsecValue = objSleep->Get(Nan::New("nsec").ToLocalChecked());
                                           ^
/Users/alexcastillo/.node-gyp/12.2.0/include/node/v8.h:3412:3: note: 'Get' has been explicitly
      marked deprecated here
  V8_DEPRECATE_SOON("Use maybe version", Local<Value> Get(Local<Value> key));
  ^
/Users/alexcastillo/.node-gyp/12.2.0/include/node/v8config.h:322:29: note: expanded from macro
      'V8_DEPRECATE_SOON'
  declarator __attribute__((deprecated(message)))
                            ^
../src/posix-clock.cpp:123:53: error: too few arguments to function call, single argument
      'context' was not specified
        sleepTimeTs.tv_sec = (time_t)secValue->Uint32Value();
                                     ~~~~~~~~~~~~~~~~~~~~~ ^
/Users/alexcastillo/.node-gyp/12.2.0/include/node/v8.h:2567:3: note: 'Uint32Value' declared here
  V8_WARN_UNUSED_RESULT Maybe<uint32_t> Uint32Value(
  ^
/Users/alexcastillo/.node-gyp/12.2.0/include/node/v8config.h:347:31: note: expanded from macro
      'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
../src/posix-clock.cpp:124:53: error: too few arguments to function call, single argument
      'context' was not specified
        sleepTimeTs.tv_nsec = (long)nsecValue->Uint32Value();
                                    ~~~~~~~~~~~~~~~~~~~~~~ ^
/Users/alexcastillo/.node-gyp/12.2.0/include/node/v8.h:2567:3: note: 'Uint32Value' declared here
  V8_WARN_UNUSED_RESULT Maybe<uint32_t> Uint32Value(
  ^
/Users/alexcastillo/.node-gyp/12.2.0/include/node/v8config.h:347:31: note: expanded from macro
      'V8_WARN_UNUSED_RESULT'
#define V8_WARN_UNUSED_RESULT __attribute__((warn_unused_result))
                              ^
../src/posix-clock.cpp:177:4: warning: 'Set' is deprecated: Use maybe version
      [-Wdeprecated-declarations]
                        AVZ_FILL_TIMESPEC(obj, remainingTimeTs.tv_sec, remainingTimeTs.tv_nsec);
                        ^
../src/posix-clock.cpp:15:13: note: expanded from macro 'AVZ_FILL_TIMESPEC'
                (target)->Set(Nan::New("sec").ToLocalChecked(), Nan::New<Number>(sec)); \
                          ^
/Users/alexcastillo/.node-gyp/12.2.0/include/node/v8.h:3358:3: note: 'Set' has been explicitly
      marked deprecated here
  V8_DEPRECATE_SOON("Use maybe version",
  ^
/Users/alexcastillo/.node-gyp/12.2.0/include/node/v8config.h:322:29: note: expanded from macro
      'V8_DEPRECATE_SOON'
  declarator __attribute__((deprecated(message)))
                            ^
../src/posix-clock.cpp:177:4: warning: 'Set' is deprecated: Use maybe version
      [-Wdeprecated-declarations]
                        AVZ_FILL_TIMESPEC(obj, remainingTimeTs.tv_sec, remainingTimeTs.tv_nsec);
                        ^
../src/posix-clock.cpp:16:13: note: expanded from macro 'AVZ_FILL_TIMESPEC'
                (target)->Set(Nan::New("nsec").ToLocalChecked(), Nan::New<Number>(nsec));
                          ^
/Users/alexcastillo/.node-gyp/12.2.0/include/node/v8.h:3358:3: note: 'Set' has been explicitly
      marked deprecated here
  V8_DEPRECATE_SOON("Use maybe version",
  ^
/Users/alexcastillo/.node-gyp/12.2.0/include/node/v8config.h:322:29: note: expanded from macro
      'V8_DEPRECATE_SOON'
  declarator __attribute__((deprecated(message)))
                            ^
../src/posix-clock.cpp:192:6: error: variable has incomplete type 'void'
void init(Handle<Object> exports) {
     ^
../src/posix-clock.cpp:192:11: error: use of undeclared identifier 'Handle'
void init(Handle<Object> exports) {
          ^
../src/posix-clock.cpp:192:18: error: 'Object' does not refer to a value
void init(Handle<Object> exports) {
                 ^
/Users/alexcastillo/.node-gyp/12.2.0/include/node/v8.h:3356:17: note: declared here
class V8_EXPORT Object : public Value {
                ^
../src/posix-clock.cpp:192:26: error: use of undeclared identifier 'exports'
void init(Handle<Object> exports) {
                         ^
../src/posix-clock.cpp:192:34: error: expected ';' after top level declarator
void init(Handle<Object> exports) {
                                 ^
                                 ;
8 warnings and 12 errors generated.
make: *** [Release/obj.target/posix-clock/src/posix-clock.o] Error 1
```

Any advice?

**Comment by avz** [comment]
I just released v2.0.0, which is migrated to the N-API and should work on all modern versions.
Thanks for the report!

**Comment by alexcastillo** [comment]
Wow - that was fast! Thank you so much!
